### ISSUE TITLE
Opendht executing basic example program from VS2017

### ISSUE THREAD

**Issue by nadendlaprasad** [savoirfairelinux/opendht#335]
While I am building openDHT basic example program in VS 2017 It is showing up the following error message in the microsoft random.h file

` Error message:   note: char, signed char, unsigned char, int8_t, and uint8_t are not allowed .
From the file: **c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.15.26726\include\random** for the code _RNG_REQUIRE_INTTYPE(uniform_int_distribution, _Ty);  `

any idea why it's occurring ?

**Comment by nadendlaprasad** [comment]
Never mind,I fixed this issue.  this is happening as VC++ not  expecting the argument type of `unit_8 `for `uniform_int_distribution` .

**Comment by aberaud** [comment]
Hi,
How did you fixed the issue ? Would you mind making a pull request ?
Best regards

**Comment by nadendlaprasad** [comment]
I have faced this issue when i try to use Linux opendht compiled files on windows. I have downloaded a fresh copy of opendht and build the project using VS2017. Once i map all the dependencies i could able to build the opendht project successfully . Please do let me know if you want me to share this(VS2017) version of opendht

**Comment by nadendlaprasad** [comment]
Facing the same issue when referring opendht project in other cpp projects.. Here the output log for more information

```
1>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.15.26726\include\random(2401): error C2338: invalid template argument for uniform_int_distribution: N4659 29.6.1.1 [rand.req.genl]/1e requires one of short, int, long, long long, unsigned short, unsigned int, unsigned long, or unsigned long long
1>c:\prasad\stellar\workspace\git\shared\opendht\crypto.h(513): note: see reference to class template instantiation 'std::uniform_int_distribution<unsigned char>' being compiled
1>c:\prasad\stellar\workspace\git\shared\opendht\crypto.h(577): note: see reference to class template instantiation 'dht::crypto::secure_vector<T>' being compiled
1>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.15.26726\include\random(2401): error C2338: note: char, signed char, unsigned char, int8_t, and uint8_t are not allowed
```

**Comment by aberaud** [comment]
Should be fixed by 897b451f446d73c940fc48b6b32e23d36426b552
